mirte_robot/robot.py

```diff
@@ -1,41 +1,94 @@
 #!/usr/bin/env python
+import os
+import platform
 import time
-import rospy
-import rosservice
-import signal
-import sys
 import math
-import atexit
+from typing import TYPE_CHECKING, Optional
+
+import rclpy
+import rclpy.node
+from rclpy.validate_namespace import validate_namespace
+
+if TYPE_CHECKING:
+    import rclpy.client
+
+from controller_manager_msgs.srv import SwitchController
+from mirte_msgs.srv import (
+    GetColorHSL,
+    GetColorRGBW,
+    GetEncoder,
+    GetIntensity,
+    GetIntensityDigital,
+    GetKeypad,
+    GetPinValue,
+    GetRange,
+    SetMotorSpeed,
+    SetOLEDFile,
+    SetOLEDText,
+    SetPinValue,
+    SetServoAngle,
+)
+from rcl_interfaces.srv import ListParameters
 
-# TODO: check if we need the telemetrix version of this?
-#sys.path.append('/usr/local/lib/python2.7/dist-packages/PyMata-2.20-py2.7.egg')  # Needed for jupyter notebooks
-#sys.path.append('/usr/local/lib/python2.7/dist-packages/pyserial-3.4-py2.7.egg')
-
-import message_filters
-from geometry_msgs.msg import Twist
-from std_msgs.msg import Int32
-from std_msgs.msg import String
-from std_msgs.msg import Empty
-from mirte_msgs.msg import *
+mirte = {}
 
-from mirte_msgs.srv import *
-from std_srvs.srv import *
+# No QoS Profiles are set, but this might not be required, since they might already behave like ROS 1 persistant.
 
-mirte = {}
 
-class Robot():
-    """ Robot API
+class Robot:
+    """Robot API
 
-        This class allows you to control the robot from Python. The getters and setters
-        are just wrappers calling ROS topics or services.
+    This class allows you to control the robot from Python. The getters and setters
+    are just wrappers calling ROS topics or services.
     """
 
+    # Implementation Notes:
+    # This class creates a hidden ROS Node for the communication.
+    # This should only be run once, however this can not be prevented from the web interface.
+    # Therefore the node is also anonymized with the current time.
+
+    def __init__(
+        self, machine_namespace: Optional[str] = None, hardware_namespace: str = "io"
+    ):
+        """Intialize the Mirte Robot API
+
+        Parameters:
+            machine_namespace (Optional[str], optional): The Namespace from '/' to the ROS namespace for the specific Mirte. Defaults to "/{HOSTNAME}". (This only has to be changed when running the Robot API from a different machine directly. It is configured correctly for the Web interface)
+            hardware_namespace (str, optional): The namespace for the hardware peripherals. Defaults to "io".
+        """
+        self._machine_namespace = (
+            machine_namespace
+            if machine_namespace and validate_namespace(machine_namespace)
+            else "/" + platform.node().replace("-", "_").lower()
```

Comment:
Maybe we should think of a way to get the namespace without using platform. We can not use platform in the web interface.